adapters/orbidder/orbidder.go

```diff
@@ -0,0 +1,139 @@
+package orbidder
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+	"net/http"
+)
+
+type OrbidderAdapter struct {
+	endpoint string
+}
+
+// MakeRequests makes the HTTP requests which should be made to fetch bids from orbidder.
+func (rcv *OrbidderAdapter) MakeRequests(request *openrtb.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var errs []error
+	var validImps []openrtb.Imp
+
+	// check if imps exists, if not return error and do send request to orbidder.
+	if len(request.Imp) == 0 {
+		return nil, []error{&errortypes.BadInput{
+			Message: "No impressions in request",
+		}}
+	}
+
+	// validate imps
+	for _, imp := range request.Imp {
+		if err := preprocess(&imp); err != nil {
+			errs = append(errs, err)
+			continue
+		}
+		validImps = append(validImps, imp)
+	}
+
+	if len(validImps) == 0 {
+		return nil, errs
+	}
+
+	//set imp array to only valid imps
+	request.Imp = validImps
+
+	requestBodyJSON, err := json.Marshal(request)
+	if err != nil {
+		errs = append(errs, err)
+		return nil, errs
+	}
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+
+	return []*adapters.RequestData{{
+		Method:  "POST",
+		Uri:     rcv.endpoint,
+		Body:    requestBodyJSON,
+		Headers: headers,
+	}}, errs
+}
+
+func preprocess(imp *openrtb.Imp) error {
+	var bidderExt adapters.ExtImpBidder
+	if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
+		return &errortypes.BadInput{
+			Message: err.Error(),
+		}
+	}
+
+	var orbidderExt openrtb_ext.ExtImpOrbidder
+	if err := json.Unmarshal(bidderExt.Bidder, &orbidderExt); err != nil {
+		return &errortypes.BadInput{
+			Message: "Wrong orbidder bidder ext: " + err.Error(),
+		}
+	}
+
+	if orbidderExt.AccountId == "" {
+		return &errortypes.BadInput{
+			Message: "Wrong orbidder bidder ext, no account id",
+		}
+	}
+
+	if orbidderExt.PlacementId == "" {
+		return &errortypes.BadInput{
+			Message: "Wrong orbidder bidder ext, no placement id",
+		}
+	}
```

Comment:
Given that these 2 parameters were marked as `required` in file static/bidder-params/orbidder.json and that prebid-server runs a parameter validation before getting to this part of the code, it is safe to assume the true branches of the `if` statements in lines 78 and 84 will be unreachable in production. We could discard those statements entirely.

Reply:
Thanks, i will drop the if statements and the tests for this cases.

Reply:
if conditions and not more needed tests are removed